exclude-gitignore does not read .git/info/exclude

<!--
If you're not sure whether what you're experiencing is a mypy bug, please see the "Question and Help" form instead.
Please consider:

- checking our common issues page: https://mypy.readthedocs.io/en/stable/common_issues.html
- searching our issue tracker: https://github.com/python/mypy/issues to see if it's already been reported
- asking on gitter chat: https://gitter.im/python/typing
-->

**Bug Report**

<!--
If you're reporting a problem with a specific library function, the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues

If the project you encountered the issue in is open source, please provide a link to the project.
-->

When using `exclude-gitignore` option then mypy ignores additional ignore rules added to `.git/info/exclude`

**To Reproduce**

```python
mkdir test-prj
cd test-prj
git init

echo '*' >> .git/info/exclude

echo 'invalid code' > main.py
mypy --exclude-gitignore .
```

**Expected Behavior**

<!--
How did you expect mypy to behave? It’s fine if you’re not sure your understanding is correct.
Write down what you thought would happen. If you expected no errors, delete this section.
-->

Mypy should respond with

```
There are no .py[i] files in directory '.'
```

as all files are excluded.

**Actual Behavior**

<!-- What went wrong? Paste mypy's output. -->

Mypy tries to read main.py

```
main.py:1: error: Invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 1.19.1
- Mypy command-line flags: exclude-gitignore
- Mypy configuration options from `mypy.ini` (and other config files): _none_
- Python version used: 3.14.3

<!-- You can freely edit this text, please remove all the lines you believe are unnecessary. -->

--- a/mypy/modulefinder.py
+++ b/mypy/modulefinder.py
@@ -728,6 +728,14 @@ def find_gitignores(dir: str) -> list[tuple[str, PathSpec]]:
     parent_dir = os.path.dirname(dir)
     if parent_dir == dir or os.path.exists(os.path.join(dir, ".git")):
         parent_gitignores = []
+        git_info_exclude = os.path.join(dir, ".git", "info", "exclude")
+        if os.path.isfile(git_info_exclude):
+            with open(git_info_exclude) as f:
+                exclude_lines = f.readlines()
+            try:
+                parent_gitignores = [(dir, PathSpec.from_lines("gitignore", exclude_lines))]
+            except GitIgnorePatternError:
+                print(f"error: could not parse {git_info_exclude}", file=sys.stderr)
     else:
         parent_gitignores = find_gitignores(parent_dir)
 

--- a/test-data/unit/cmdline.test
+++ b/test-data/unit/cmdline.test
@@ -989,6 +989,17 @@ b
 [out]
 b/bpkg.py:1: error: "int" not callable
 
+[case testCmdlineExcludeGitignoreWithGitInfoExclude]
+# cmd: mypy --exclude-gitignore .
+[file .git/info/exclude]
+abc
+[file abc/apkg.py]
+1()
+[file c/cpkg.py]
+1()
+[out]
+c/cpkg.py:1: error: "int" not callable
+
 [case testCmdlineCfgExclude]
 # cmd: mypy .
 [file mypy.ini]

